lib/data/network_services/mocked_network_service.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:spotify_prototype/data/models/track_model.dart';
+
+abstract class NetworkService {
+  Future<List<TrackModel>> fetchModel({
+    required String search,
+    required int limit,
+  });
+}
+
+class MockedServiceImpl extends NetworkService {
```

Comment:
i would like to see mocked in different file

Reply:
@k1mka i prefer next structure
network service
mocked network service
network service impl

each class in diffener file